api/tests/routes/pro/patch_offerer_bank_accounts_test.py

```diff
@@ -0,0 +1,524 @@
+                    type=history_models.ActionType.LINK_VENUE_BANK_ACCOUNT_DEPRECATED,
+                    userId=pro_user.id,
+                    venueId=third_venue.id,
+                    offererId=offerer.id,
+                    bankAccountId=bank_account.id,
+                ),
+            ]
+        )
+
+        assert response.status_code == 204
+
+        response = http_client.get(f"/offerers/{offerer.id}/bank-accounts/")
+
+        assert response.status_code == 200
+        assert len(response.json["bankAccounts"]) == 1
+        bank_account_response = response.json["bankAccounts"].pop()
+        assert not bank_account_response["linkedVenues"]
+
+        db_session.refresh(bank_account)
+
+        assert len(bank_account.venueLinks) == 3
+
+        assert (
+            not offerers_models.VenueBankAccountLink.query.join(finance_models.BankAccount)
+            .filter(
+                finance_models.BankAccount.id == bank_account.id,
+                offerers_models.VenueBankAccountLink.timespan.contains(datetime.datetime.utcnow()),
+            )
+            .count()
+        )
+
+        # For compliance/fraud reasons, we want to make sure those actions are logged.
+        actions_logged = (
+            history_models.ActionHistory.query.filter(
+                history_models.ActionHistory.actionType.in_(
+                    (
+                        history_models.ActionType.LINK_VENUE_BANK_ACCOUNT_CREATED,
+                        history_models.ActionType.LINK_VENUE_BANK_ACCOUNT_DEPRECATED,
+                    )
+                )
+            )
+            .order_by(history_models.ActionHistory.actionDate, history_models.ActionHistory.venueId)
+            .all()
+        )
+
+        assert len(actions_logged) == len(actions_occured)
+
+        for action_logged, action_occured in zip(actions_logged, sorted(actions_occured, key=lambda a: a.venueId)):
+            assert action_logged.actionType == action_occured.type
+            assert action_logged.userId == action_occured.userId
+            assert action_logged.venueId == action_occured.venueId
+            assert action_logged.offererId == action_occured.offererId
+            assert action_logged.bankAccountId == action_occured.bankAccountId
+
+    def test_adding_new_venue_link_doesnt_alter_historic_links(self, db_session, client):
+        actions_occured = []
+
+        offerer = offerers_factories.OffererFactory()
+        pro_user = users_factories.ProFactory()
+        offerers_factories.UserOffererFactory(user=pro_user, offerer=offerer)
+        bank_account = finance_factories.BankAccountFactory(offerer=offerer)
+        first_venue = offerers_factories.VenueFactory(managingOfferer=offerer)
+        second_venue = offerers_factories.VenueFactory(managingOfferer=offerer)
+        third_venue = offerers_factories.VenueFactory(managingOfferer=offerer)
+        fourth_venue = offerers_factories.VenueFactory(managingOfferer=offerer)
```

Comment:
Suggestion : pour éviter la répétition, tu peux écrire : 
```suggestion
        venues = offerers_factories.VenueFactory.create_batch(4, managingOfferer=offerer)
```